app/models/concerns/node_shared.rb

```diff
@@ -8,6 +8,14 @@ def likes
   def liked_by(uid)
     likers.collect(&:uid).include?(uid)
   end
+  
+  def self.button(body)
+    body.gsub(/(?<![\>`])(\<p\>)?\[button\:(.+)\:(\S+)\]/) do |_tagname|
+      btnText = Regexp.last_match(2)
+      btnHref = Regexp.last_match(3)
+      return "<a class='btn btn-primary' href=" + btnHref + "'>" + btnText + "</a>"
```

Comment:
@jywarren found it! I say that it expected 1 but got zero, so I knew there was an issue with line 28 or 29 in node_shared_test. I wrote href=' then ". I removed that single quote. Should work now.